lib/android/app/src/main/java/com/reactnativenavigation/views/stack/topbar/titlebar/MainToolBar.kt

```diff
@@ -160,6 +142,69 @@ class MainToolBar(context: Context) : RelativeLayout(context) {
         clearComponent()
     }
 
+    override fun onLayout(changed: Boolean, l: Int, t: Int, r: Int, b: Int) {
+        super.onLayout(changed, l, t, r, b)
+        // Bypass regular layout for the titleComponent to make final placements based on the measurement
+        val titleComponent = getTitleComponent()
+        if (titleComponentAlignment == Alignment.Center) {
+            val titleLeft = max(l + leftButtonsBar.measuredWidth, (r - l) / 2 - titleComponent.measuredWidth / 2)
+            val titleRight = min(r - rightButtonsBar.measuredWidth, (r - l) / 2 + titleComponent.measuredWidth / 2)
+            titleComponent.layout(titleLeft, t, titleRight, b)
+        } else {
+            titleComponent.layout(l + leftButtonsBar.measuredWidth + defaultMargin, t, r - rightButtonsBar.measuredWidth, b)
+        }
+    }
+
+    override fun onMeasure(widthMeasureSpec: Int, heightMeasureSpec: Int) {
+        val mode = MeasureSpec.getMode(widthMeasureSpec)
+        val parentWidth = MeasureSpec.getSize(widthMeasureSpec)
+        val parentHeight = MeasureSpec.getSize(heightMeasureSpec)
+        logd("OnMeasure: parentWidth: $parentWidth, parentHeight: $parentHeight widthMeasureSpec:" +
```

Comment:
Don't forget to remove logs when ready. Perhaps wrap it with [Log.isLoggable](https://developer.android.com/reference/android/util/Log#isLoggable(java.lang.String,%20int)) just in case.